### PR TITLE
Issue 1296 [hotfix]: fix trufflehog scans in upstream repos

### PR DESCRIPTION
**Pull request by jorgegonzalez**
### Summary of Changes

This PR addresses a recent issue with the Trufflehog script scanning invalid branches in upstream repositories. 

Closes issue #1296 

_Acceptance criteria as stated in the issue_

### How to Test
_List the steps to test the PR_
These steps are generic, please adjust as necessary.
```
cd tdrs-frontend && docker-compose -f docker-compose.yml -f docker-compose.local.yml up -d
cd tdrs-backend && docker-compose -f docker-compose.yml -f docker-compose.local.yml up -d 
```

1. Open http://localhost:3000/ and sign in.
1. Proceed with functional tests as described herein.
1. Test steps should be captured in the demo GIF(s) and/or screenshots below.

> *Demo GIF(s) and screenshots for testing procedure*

### Deliverable 1: Accepted Features

> *Performance Standard(s): At the beginning of each sprint, the Product Owner and development team will collaborate to define a set of user stories to be completed during the sprint. Acceptance criteria for each story will also be defined. The development team will deliver code and functionality to satisfy these user stories.*

> *Acceptable Quality Level: Delivered code meets the acceptance criteria for each user story. Incomplete stories will be assessed and considered for inclusion in the next sprint.*

+ [ ] Look up the acceptance criteria in the related issue; paste ACs below in checklist format.
+ [ ] Check against the criteria:

As Product Owner, @lfrohlich will decide if ACs are met.

### Deliverable 2: Tested Code

> *Performance Standard(s): Code delivered under the order must have substantial test code coverage. Version-controlled HHS GitHub repository of code that comprises products that will remain in the government domain.*

> *Acceptable Quality Level: Minimum of 90% test coverage of all code. All areas of code are meaningfully tested.*

+ Are all areas of code introduced in this PR meaningfully tested?
  + [ ] If this PR introduces backend code changes, are they meaningfully tested?
  + [ ] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [ ] Frontend coverage: ___ (see https://github.com/raft-tech/TANF-app README coverage dashboard)
  + [ ] Backend coverage: ___ (see https://github.com/raft-tech/TANF-app README coverage dashboard)

### Deliverable 3: Properly Styled Code

> *Performance Standard(s): GSA 18F Front- End Guide*

> *Acceptable Quality Level: 0 linting errors and 0 warnings*

+ [x] Are backend code style checks passing on CircleCI?
+ [x] Are frontend code style checks passing on CircleCI?
+ [ ] Does this PR change any linting or CI settings?

### Deliverable 4: Accessible

> *Performance Standard(s): Web Content Accessibility Guidelines 2.1 AA standards*

> *Acceptable Quality Level: 0 errors reported using an automated scanner and 0 errors reported in manual testing*
+ [ ] Does this PR complete the epic? 
+ [ ] Are links included to any other gov-approved PRs associated with epic?
+ [ ] Does PR include documentation for Raft's a11y review? 
+ [ ] Did automated and manual testing with @iamjolly and @ttran-hub using Accessibility Insights reveal any errors introduced in this PR?
    + [See the full Accessibility Assessment plan here.](https://github.com/HHS/TANF-app/blob/main/docs/Technical-Documentation/how-government-will-test-a11y.md)

### Deliverable 5: Deployed

> *Performance Standard(s): Code must successfully build and deploy into the staging environment.*

> *Acceptable Quality Level: Successful deployment by assigning a 'Deploy with CircleCI - {env}' label*

+ [x] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### Deliverable 6: Documented

> *Performance Standard(s): Summary of user stories completed every two weeks. All dependencies are listed and the licenses are documented. Major functionality in the software/source code is documented, including system diagram. Individual methods are documented inline in a format that permits the use of tools such as JSDoc. All non-inherited 800-53 system security controls are documented in the Open Control or OSCAL format and HHS Section 508 Product Assessment Template (PAT) are updated as appropriate.*

> *Acceptable Quality Level: Combination of manual review and automated testing, if available*

+ [ ] If this PR introduces backend code, is that code documented both inline and overall?
+ [ ] If this PR introduces frontend code, is that code documented both inline and overall?
+ [ ] If this PR introduces dependencies, are their licenses documented?

### Deliverable 7: Secure

> *Performance Standard(s): Open Web Application Security Project (OWASP) Application Security Verification Standard 3.0*

> *Acceptable Quality Level: Code submitted must be free of medium- and high-level static and dynamic security vulnerabilities*

+ [ ] Does the OWASP Scan pass on CircleCI?
+ [ ] Do manual code review and manual testing detect any security issues?

### Deliverable 8: Context

> * Performance Standard(s): Code must be understandable and contextualized for the reviewers possess the knowledge and background necessary for analysis and constructive criticism to take place.

> * Acceptable Quality Level: Code submitted in the pull request has context.*
+ [x] Does this pull request contain sufficient inline comments providing relevant context for code snippets?
+ [x] Is the code understandable and lucid?
+ [x] Does this pull request provide background for why coding decisions were made?
+ [x] Can you as a reviewer explain and take ownership of these elements presented in this code review?

### CODE REVIEW

File: scripts/trufflehog-check.sh [hunk 1]
```diff
@@ -15,19 +15,30 @@ fi
...
 python ./trufflehog-check/lib/python3.8/site-packages/truffleHog/truffleHog.py \
   --regex \
   --entropy=True \
-  --branch "$1" \
+  --branch "$CIRCLE_BRANCH" \
   --since_commit "$last_merge" \
   --exclude_paths ./trufflehog-exclude-patterns.txt \
   --rules ./regexes.json \
-  https://github.com/raft-tech/TANF-app
+  file:///root/project
```
Comment (jorgegonzalez) [inline scripts/trufflehog-check.sh:41]: Updated this to scan the locally cloned repo, so we don't have to specify a given repository by url.

File: scripts/trufflehog-check.sh [hunk 2]
```diff
@@ -15,19 +15,30 @@ fi
...
 
-last_merge=$(git log --format=format:"%H" -n 1 raft-tdp-main)
-
 # $1 - The first argument to this script, the current git branch name
+CIRCLE_BRANCH=$1
+
+# If the current PR branch includes `raft-tdp-main`, it a merge against HHS:main,
+# so we update the target branch to scan against.
+if [ "$CIRCLE_BRANCH" == "raft-tdp-main" ] ; then
+  TARGET=main
```
Comment (jorgegonzalez) [inline scripts/trufflehog-check.sh:24]: This will set the correct target branch to scan against when merging into raft-tdp-main.

**Review by jtwillis92 — APPROVED** [review]
LGTM

**Review by abottoms-coder — APPROVED** [review]
Alright, code looks solid but let me know if we continue to experience issues on HHS @ADPennington / @jorgegonzalez

**Review by ADPennington — APPROVED** [review]
👍🏾 will test on `hhs:main`. ticked off relevant checks in PR summary.

**Pull request merged**